### PR TITLE
Remove AudioContextInfo IDL and move attributes to AudioWorkletGlobalScope

### LINKED ISSUE #1326: [audioworklet] Moving AudioContextInfo to AudioWorkletGlobalScope
Currently `AudioContextInfo` is located at `AudioWorkletProcessor`, but it seems wasteful considering `sampleRate` and `currentTime` are identical to all instances in an `AudioWorkletGlobalScope`.

So I am proposing moving individual attributes of `AudioContextInfo` to `AudioWorkletGlobalScope`:
```webidl
interface AudioWorkletGlobalScope : WorkletGlobalScope {
  void registerProcessor (DOMString name, VoidFunction processorCtor);
  readonly attribute double currentTime;
  readonly attribute float sampleRate;
}
```

Calling `AudioContextInfo()` creates a dictionary, so accessing the primitive values directly should be more efficient and less wasteful. (in terms of GC)

### PR DESCRIPTION
**Pull request by hoch**
- Fixes #1326 
- Removes `AudioContextInfo` IDL and `AudioWorkletProcessor.getContextInfo()`.
- Add `currentTime` and `sampleRate` to `AudioWorkletGlobalScope`.

***
[Preview](https://s3.amazonaws.com/pr-preview/hoch/web-audio-api/1326-remove-audiocontextinfo.html) | [Diff](https://s3.amazonaws.com/pr-preview/WebAudio/web-audio-api/e669c8f...hoch:da4b4ef.html)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -8708,10 +8708,37 @@ <h2 id="AudioWorkletGlobalScope">
...
 };
           </pre>
           <section>
             <h3>
+              Attributes
+            </h3>
+            <dl class="attributes" data-dfn-for="AudioWorkletGlobalScope"
+            data-link-for="AudioWorkletGlobalScope">
+              <dt>
+                <dfn><code>currentTime</code></dfn> of type <span class=
```
Comment (rtoy) [inline index.html]: This should be `<code><dfn>`

File: index.html [hunk 2]
```diff
@@ -8708,10 +8708,37 @@ <h2 id="AudioWorkletGlobalScope">
...
+                "idlAttrType"><code>double</code></span>, readonly
+              </dt>
+              <dd>
+                The context time of the block of audio being processed. By
+                definition this will be equal to the value of
+                <a>BaseAudioContext</a>'s <a>currentTime</a> attribute that
```
Comment (rtoy) [inline index.html]: Does the link to currentTime actually work?

File: index.html [hunk 3]
```diff
@@ -8708,10 +8708,37 @@ <h2 id="AudioWorkletGlobalScope">
...
+                was most recently observable in the <a>control thread</a>.
+              </dd>
+              <dt>
+                <dfn><code>sampleRate</code></dfn> of type <span class=
```
Comment (rtoy) [inline index.html]: Swap `<dfn><code>`.

File: index.html (hunk 2 above)
Comment (joeberkovitz) [reply index.html]: @hoch I think you need the <a data-link-for="BaseAudioContext"> qualifier here

**Comment by hoch** [comment]
I think I have a bigger problem with this PR. Somehow it fails on tidy check and I can't figure out why.

**Comment by joeberkovitz** [comment]
It looks as though the indentation changed -- maybe section tags got scrambled?

Part of the diff that Travis is complaining about:

```
-        <section>
-          <h2 id="instantiation-of-AudioWorkletNode-and-AudioWorkletProcessor">
-            The instantiation of <a>AudioWorkletNode</a> and
-            <a>AudioWorkletProcessor</a>
-          </h2>
-          <p>
-            When the constructor of <a>AudioWorkletNode</a> is invoked in the
-            main global scope, the corresponding <a>AudioWorkletProcessor</a>
-            instance is automatically created in
-            <a>AudioWorkletGlobalScope</a>. After the construction, they
-            maintain the internal reference to each other until the
-            <a>AudioWorkletNode</a> instance is destroyed.
-          </p>
+            <h2 id=
+            "instantiation-of-AudioWorkletNode-and-AudioWorkletProcessor">
+              The instantiation of <a>AudioWorkletNode</a> and
+              <a>AudioWorkletProcessor</a>
+            </h2>
+            <p>
+              When the constructor of <a>AudioWorkletNode</a> is invoked in the
+              main global scope, the corresponding <a>AudioWorkletProcessor</a>
+              instance is automatically created in
+              <a>AudioWorkletGlobalScope</a>. After the construction, they
+              maintain the internal reference to each other until the
+              <a>AudioWorkletNode</a> instance is destroyed.
+            </p>
```

**Comment by hoch** [comment]
@rtoy @joeberkovitz Thanks for the feedback! PTAL again.

File: index.html [hunk 4]
```diff
@@ -9374,8 +9375,8 @@ <h2 id="defining-a-valid-audioworkletprocessor">
                 this case, <code>process()</code> SHOULD return
                 <code>true</code> for some period of time after
                 <code>inputs</code> is found to contain zero channels. The
-                current time may be obtained from the processor's
-                <code>contextInfo</code> to measure the start and end of this
+                current time may be obtained from the global scope's
+                <code>currentTime</code> to measure the start and end of this
```
[the remark this answers is not in the document]
Comment (hoch) [reply index.html]: You meant `<a data-link-for="AudioWorkletGlobalScope">currentTime</a>`?

[the remark this answers is not in the document]
Comment (rtoy) [reply index.html]: Yeah.

**Comment by hoch** [comment]
@joeberkovitz Can you take a look?

**Pull request merged**